In [4]:
import numpy as np

A = np.array([[56.0, 0.0, 4.4, 68.0],
              [1.2, 104.0, 52.0, 8.0],
              [1.8, 135.0, 99.0, 0.9]])

print(A)

[[ 56.    0.    4.4  68. ]
 [  1.2 104.   52.    8. ]
 [  1.8 135.   99.    0.9]]


This columns in this matrix are types of food and the rows are the number of calories per 100g of food are carbs (top row), protein (middle), or fat (bottom)

In [5]:
cal = A.sum(axis = 0)

print(cal)

[ 59.  239.  155.4  76.9]


cal represtents the total number of calories for each food

In [7]:
percentage = 100*A/cal.reshape(1,4)
print(percentage)

[[94.91525424  0.          2.83140283 88.42652796]
 [ 2.03389831 43.51464435 33.46203346 10.40312094]
 [ 3.05084746 56.48535565 63.70656371  1.17035111]]


This matrix represents the % breakdown of calories for each food
E.g. The first food is 94.9% carbs, 2% protein and 3.1% fat